ProUI feature utilization / Misc. Fixes

classicrocker883 opened the pull request:
<!--

Submitting a Pull Request

- Please fill out all sections of this form. You can delete the helpful comments.
- Pull Requests without clear information will take longer and may even be rejected.
- We get a high volume of submissions so please be patient during review.

-->

### Description
This allows the User to easily enable features of the ProUI menu.  
The **#define** of some features has not yet been included.  
I moved these here into **lcd/e3v2/proui/dwin_defines.h** file because that seemed most appropriate.

---

there is a warning after commit 376673d "🚸 Minor touch calibration improvement (#26445)"
```r
Marlin\src\lcd\e3v2\proui\bedlevel_tools.cpp: In static member function 'static void BedLevelTools::drawBedMesh(int16_t, uint8_t, uint16_t, uint16_t)':
Marlin\src\lcd\e3v2\proui\bedlevel_tools.cpp:250:50: warning: narrowing conversion of '(((int)(((short unsigned int)((int)cell_height_px)) / 2)) - ((int)fs))' from 'int' to 'signed char' [-Wnarrowing]
  250 |         xy_int8_t offset { 0, cell_height_px / 2 - fs };
```

I simply changed it to `long`
```diff
-xy_int8_t offset { 0, cell_height_px / 2 - fs };
+xy_long_t offset { 0, cell_height_px / 2 - fs };
```
and this stopped the warning. and also for JyersUI.

Update
---
for the commit "indent" I just indented the files to be inline with `#if` statements

in **dwin.cpp**, I removed `onDrawManualTramming()` for it has no use  
and change `#if` statements from "&&" to ALL, and "||" to ANY

then I updated **dwin_defines.h** to remove `PROUI_PID_TUNE` from Conditionals_post.h

also, I renamed `UBL_Z_OFFSET_MIN/MAX` to just `Z_OFFSET_MIN/MAX`, and removed the redundant `#define Z_OFFSET_xxx` from **dwin.cpp**
<!--

Clearly describe the submitted changes with lots of details. Include images where helpful. Initial reviewers may not be familiar with the subject, so be as thorough as possible. You can use MarkDown syntax to improve readability with bullet lists, code blocks, and so on. PREVIEW and fix up formatting before submitting.

-->

### Requirements
`DWIN_LCD_PROUI`
<!-- Does this PR require a specific board, LCD, etc.? -->

### Benefits
makes it easier to enable/disable ProUI menu features.
<!-- What does this PR fix or improve? -->

### Configurations

<!-- Attach Configurations ZIP and any other files needed to test this PR. -->

### Related Issues
~~There was difficulty having the compiler include the file for the **#define** values.~~

~~So I added `#include "../lcd/e3v2/proui/dwin_defines.h"` into **MarlinConfigPre.h**, and that solved it. I suppose I could have added everything to the **Configuration.h** file under where `DWIN_LCD_PROUI` is *defined*...~~

~~but as I said, having it in **dwin_defined.h** is logical.~~
<!-- Does this PR fix a bug or fulfill a Feature Request? Link related Issues here. -->

classicrocker883 committed:
Fix typo in stm32f4.ini

classicrocker883 committed:
Add ProUI feature options
The following file changes allows fuller utilization of ProUI features.

dwin.h
dwin_defines.h
SanityCheck.h
MarlinConfigPre.h

classicrocker883 committed:
fix `SDSORT_GCODE false` warning

classicrocker883 commented on Marlin/src/lcd/e3v2/proui/dwin.h:
```diff
@@ -39,13 +39,6 @@
   #include "../../../feature/leds/leds.h"
 #endif
 
-#if ANY(BABYSTEPPING, HAS_BED_PROBE)
```
to clarify this change, this is redundant, so it removed.

classicrocker883 committed:
include #ifdef in MarlinConfigPre.h
for DWIN_LCD_PROUI

thinkyhead commented on Marlin/src/inc/MarlinConfigPre.h:
```diff
@@ -60,3 +60,7 @@
+
+#ifdef DWIN_LCD_PROUI
+  #include "../lcd/e3v2/proui/dwin_defines.h"
+#endif
```
It would be better if this was not here. This include should just be added to whatever CPP and H files specifically need it.

thinkyhead committed:
enabled

thinkyhead committed:
Merge branch 'bugfix-2.1.x' into pr/26434

thinkyhead committed:
tweak errors

thinkyhead committed:
clarify

thinkyhead commented:
In general, ProUI and other LCD implementation should be using the existing Marlin settings instead of adding their own. For example, ProUI should be using the `LCD_BED_LEVELING` flag to decide whether or not to display bed leveling options, and the standard `MESH_EDIT_MENU` setting to determine whether or not to show its own mesh editor. At the moment, ProUI displays the mesh edit menu whenever there is a mesh, and it ignores the `LCD_BED_LEVELING` setting, so it needs some more work to get into compliance.

thinkyhead committed:
Merge branch 'bugfix-2.1.x' into pr/26434

classicrocker883 committed:
Fix warning: narrowing conversion
introduced in commit "🚸 Minor touch calibration improvement (#26445)"

thinkyhead committed:
fix file wrappers

classicrocker883 committed:
Merge branch 'bugfix-2.1.x-Nov3' of https://github.com/classicrocker883/Marlin into bugfix-2.1.x-Nov3

classicrocker883 committed:
Fix warning: narrowing conversion in JyersUI
introduced in commit "🚸 Minor touch calibration improvement (#26445)"

classicrocker883 committed:
fixes for dwin.cpp
onDrawManualTramming() has no use - removed
change "&&" to ALL, and "||" to ANY

classicrocker883 commented on Marlin/src/lcd/e3v2/proui/dwin_defines.h:
```diff
@@ -78,3 +78,29 @@
+#if HAS_MESH
+  #define PROUI_MESH_EDIT   // Add a menu to edit mesh points
+#endif
+#if ANY(PROUI_PID_TUNE, MPC_AUTOTUNE) && DISABLED(DISABLE_TUNING_GRAPH)
```
unless `#include "../../../inc/Conditionals_post.h"` is included, `PROUI_PID_TUNE` isn't seen as defined, and so `PROUI_TUNING_GRAPH` is not enabled

thinkyhead replied:
Right. Any header that refers to settings needs to include `MarlinConfigPre.h` so that is the solution I just applied.

classicrocker883 commented on Marlin/src/lcd/e3v2/proui/dwin_defines.h:
```diff
@@ -78,3 +78,29 @@
+#if HAS_MESH
+  #define PROUI_MESH_EDIT   // Add a menu to edit mesh points
+#endif
+#if ANY(PROUI_PID_TUNE, MPC_AUTOTUNE) && DISABLED(DISABLE_TUNING_GRAPH)
```
having DISABLED(DISABLE_TUNING_GRAPH) is kind of redundant... there is no other reference to this.
instead if a user wants to disable the PID graph, they can simply comment `PROUI_TUNING_GRAPH` out.
```suggestion
#if ANY(PROUI_PID_TUNE, MPC_AUTOTUNE)
```

thinkyhead replied:
This is a hidden option for developer and troubleshooting use. It should be left in.

classicrocker883 committed:
indent

classicrocker883 committed:
add brief and param to Menu Line Spacing

classicrocker883 committed:
update dwin_defines.h
remove PROUI_PID_TUNE from Conditionals_post.h

thinkyhead committed:
undo extraneous changes

classicrocker883 committed:
rename UBL_Z_OFFSET_MIN/MAX to Z_OFFSET_MIN/MAX

classicrocker883 committed:
Merge branch 'bugfix-2.1.x-Nov3' of https://github.com/classicrocker883/Marlin into bugfix-2.1.x-Nov3

thinkyhead commented:
General code notes:
- Since `PROUI_PID_TUNE` is used to enable some "global" behavior it's fine to leave it in `Conditionals_post.h` for now. It may become a standard to include LCD files where they are referenced in the general codebase, but it makes sense first to extend ExtUI and add ExtUI interfaces to displays that have special handling in the general codebase, such as the ProUI.
- Indentation doesn't need to be applied to the first level when "the whole file" is wrapped in `#if`/`#endif` as a unit. If there are two blocks of equal length split by an `#else` we might indent a long file.
- Conditionals like `HAS_MESH` can "hang loose" when used in combination, and don't need to be wrapped in `ANY()` or `ALL()`. For a situation like `ENABLED(BLAH) && HAS_FOO`, it is ok to combine as `ALL(BLAH, HAS_FOO)` for brevity.
- Doxygen / Headerdoc recognize end-of-line comments starting with `//<` or `//<!` as documentation. EOL comments are preferred for documenting members of structs and enums.

classicrocker883 committed:
forgot indent bedlevel_tools.cpp

thinkyhead committed:
unindent

thinkyhead committed:
move to dwin_defines

thinkyhead commented:
Seems pretty complete to me. Will merge shortly….

thinkyhead committed:
rgb type, some case fix
